src/Console/NotifyCommand.php

```diff
@@ -102,7 +103,17 @@ public function handle()
 
         $errors = [];
 
-        $query->chunkById(20, function ($discussions) use (&$errors) {
+        $query->chunkById(20, function ($discussions) use ($canSelectOwn, &$errors) {
+            // Filter out discussions where the user can't select a post as best answer.
+            // - The user must have permission to select a best answer on their own discussion
+            // - The user must be able to select a post, whether they can select any post (including their own) or not.
+            $discussions = $discussions->filter(function ($d) use ($canSelectOwn) {
+                $hasPermission = $d->user->can('selectBestAnswerOwnDiscussion', $d);
+                $canSelectPosts = $canSelectOwn || $d->posts()->where('user_id', '!=', $d->user_id)->count() != 0;
```

Comment:
This doesn't use the repository because then I'd have to loop through all the posts, run the can select post as best answer method on each. This essentially does the same (except I guess it doesn't account for event posts, but it's an improvement over the current behavior).